aiida_workgraph/web/frontend/src/components/DataNodeTable.js

```diff
@@ -107,8 +110,14 @@ function DataNode() {
                             <td>{item.node_type}</td>
                             <td>{item.label}</td>
                             <td>
-                                <button onClick={() => handleDeleteClick(item)} className="action-button delete-button"><FaTrash /></button>
+                                <button onClick={() => setModalShow(true)} className="action-button delete-button"><FaTrash /></button>
                             </td>
+                            <WorkGraphDeleteNodePrompt
+                                show={modalShow}
+                                item={item}
+                                onYesClick={() => handleDeleteConfirmClick(item)}
+                                onNoClick={() => setModalShow(false)}
+                            />
```

Comment:
There are two issues here:

- You set ModalShow to true, this will affect all modal, since you added modal inside each row.
- Better not add a modal inside each row. You can add a new state parameter, `item,` to record the row to be deleted.

Reply:
Thanks! now we use a state variable to set the item plus useEffect to detect the change of the state variable that contains the state.